numpy/linalg/linalg.py

```diff
@@ -1142,6 +1144,8 @@ def eig(a):
     _assertNdSquareness(a)
```

Comment:
This would previously produce a cryptic `ValueError: cannot remove a zero-sized axis from an iterator`, since it omitted the check in other places.

It's not clear to me why `NpyIter_RemoveAxis` does not allow a zero-sized axis to be removed. At the very least, it should allow all but the last zero-sized axis to be removed, I think?